src/web/app/components/question-submission-form/question-submission-form.component.ts

```diff
@@ -197,4 +196,19 @@ export class QuestionSubmissionFormComponent implements OnInit {
     return this.feedbackResponseService.isFeedbackResponseDetailsEmpty(
         this.model.questionType, responseDetails);
   }
+
+  /**
+   * Updates validity of all responses in a question.
+   */
+  updateValidity(isValid: boolean): void {
+    const recipientSubmissionForms: FeedbackResponseRecipientSubmissionFormModel[] =
+        this.model.recipientSubmissionForms.slice();
```

Comment:
This is shallow copy actually.

Reply:
I just took this from `triggerRecipientSubmissionFormChange()` haha, would you like me to change that method too?

Reply:
That method is actually doing thing correctly :P

This is the trick:
https://github.com/TEAMMATES/teammates/blob/f48c9579da351f9bc68edede51186e004208eb11/src/web/app/components/question-submission-form/question-submission-form.component.ts#L138-L141